tests/unit/test-extension-runners/test.extension-runners.js

```diff
@@ -368,6 +405,24 @@ describe('util/extension-runners', () => {
       );
     });
 
+    it('lints changed files', () => {
```

Comment:
we should also add some test cases for the failures scenarios (e.g. linting failure when the first linting has been executed while the extension runner is starting, linting failures on source change) and ensure that it is going to behave like we want (e.g. the linting failure on the first linting could be a fatal error, and exit the web-ext run command, while a linting failure on change could send a desktop notification and it does not reload the changed extension, this sounds like a reasonable behavior to me)